### PR TITLE
Hooks - useWalletConnect - unit tests 

### PR DESCRIPTION
**Pull request by alvaro-ricotta**
## Testing

image

- Layer L1/L2 should triggerInit if accountEnable is false and baseEnabled is true
- Layer L1/L2 should triggerInit if accountEnable is true and baseEnabled is true but chain has been changed
- triggerInit should Open No Metamask Modal when intialized === nometamask
- triggerInit should Open Wrong Network Modal when intialized === wrongnetwork
- triggerInit should Set account enabled False when intialized === false
- triggerInit should Close wrong network modal , setLayer , setEnabledAccout, setWalletAddress and addTooken when initialized === L1 or L2
- triggerInit should return false when intialized is not defined
- connectBOBARequest should tigger doConnectToLayer(L2), reset chain Connection and open walletSelectorModal when provider is not defined
- connectETHRequest should tigger doConnectToLayer(L1), reset chain Connection and open walletSelectorModal

### CODE REVIEW

File: src/hooks/useWalletConnect/index.ts [hunk 1]
```diff
@@ -132,19 +133,24 @@ export const useWalletConnect = () => {
   // listening for l1 connection request
   useEffect(() => {
     if (connectETHRequest) {
+      console.log('insideConnectEthRequest')
```
Comment (sk-enya) [inline src/hooks/useWalletConnect/index.ts:136]: please remove console.

File: src/hooks/useWalletConnect/index.ts [hunk 2]
```diff
@@ -132,19 +133,24 @@ export const useWalletConnect = () => {
...
+
       doConnectToLayer('L1')
     }
   }, [connectETHRequest, doConnectToLayer])
 
   // listening for l2 connection request
   useEffect(() => {
     if (connectBOBARequest) {
+      console.log('inside connect boba request')
```
Comment (sk-enya) [inline src/hooks/useWalletConnect/index.ts:145]: remove consoles!

File: src/hooks/useWalletConnect/index.ts [hunk 3]
```diff
@@ -132,19 +133,24 @@ export const useWalletConnect = () => {
...
       doConnectToLayer('L2')
     }
   }, [connectBOBARequest, doConnectToLayer])
 
   useEffect(() => {
     if (connectRequest && !networkService.walletService.provider) {
+      console.log('DISABLE_WALLETCONNECT', DISABLE_WALLETCONNECT)
+      console.log('inside connectRequest')
```
Comment (sk-enya) [inline src/hooks/useWalletConnect/index.ts:153]: remove consoles!

File: src/util/tests/index.ts [hunk 4]
```diff
@@ -71,3 +71,7 @@ export const mockLocalStorage = (() => {
     },
   }
 })()
+
+export type EnvType = string | number | null | undefined
+
+export const DISABLE_WALLETCONNECT: EnvType = 'true'
```
Comment (sk-enya) [inline src/util/tests/index.ts:77]: Do we really need this? as it's seems duplicate declaration with constant.ts. 

Please add comments so we can refer in future to understand correctly about why we have written this again?

**Review by sk-enya — CHANGES_REQUESTED** [review]
Please follow the PR structure for descriptions. you can inject as much details and helpful for reviewer to read it correctly.
Screenshot 2023-11-24 at 1 06 05 AM

File: src/util/tests/index.ts (hunk 4 above)
Comment (alvaro-ricotta) [reply src/util/tests/index.ts]: removed

**Review by alvaro-ricotta — COMMENTED** [review]
all task has been resolved

**Review by sk-enya — APPROVED** [review]

**Pull request merged**